File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(long long x) {
        long long OriginalX=x;
        long long rev=0;
         x=std::abs(OriginalX);
        while(x>0)
        {
            int rem=x%10;
            rev=(10*rev)+rem;
            x=x/10;
        }
        if(rev<INT_MIN||rev>INT_MAX)
        {
            return 0;
        }
        if(OriginalX<0)
        {
            return -rev;
        
        }
        else
        {
            return rev;
        }
    }
};

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        OriginalX=x
        x=abs(OriginalX)
        rev=int(str(x)[::-1])
        if rev<-2147483648 or rev>2147483647:
            return 0
        if (OriginalX>0):
            return rev
        else:
            return -rev
        